specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2021-05-13-preview/desktopvirtualization.json

```diff
@@ -0,0 +1,6389 @@
+              }
+            ]
+          },
+          "type": "string",
+          "description": "Action types for controlling hostpool update.",
+          "x-nullable": false
+        }
+      }
+    },
+    "HostPoolUpdateProgress": {
+      "description": "Properties of host pool update progress.",
+      "type": "object",
+      "readOnly": true,
+      "properties": {
+        "timeCreated": {
+          "format": "date-time",
+          "type": "string",
+          "description": "The hostpool update create time."
+        },
+        "timeStarted": {
+          "format": "date-time",
+          "type": "string",
+          "description": "The hostpool update start time."
+        },
+        "timeFailed": {
+          "format": "date-time",
+          "type": "string",
+          "description": "The hostpool update fail time."
+        },
+        "timeEnded": {
+          "format": "date-time",
+          "type": "string",
+          "description": "The hostpool update end time."
+        },
+        "listOfErrors": {
+          "type": "array",
+          "description": "The alerts given to customers for hostpool update.",
+          "items": {
+            "$ref": "#/definitions/HostPoolUpdateFault"
+          }
+        },
+        "progressPercentage": {
+          "type": "number",
+          "format": "float",
+          "description": "The progress percentage."
+        },
+        "sessionHostsToMigrate": {
+          "type": "integer",
+          "format": "int32",
+          "description": "The number of session hosts to migrate by hostpool update."
+        },
+        "sessionHostsMigrating": {
+          "type": "integer",
+          "format": "int32",
+          "description": "The number of session hosts migrating by hostpool update."
+        },
+        "sessionHostsMigrated": {
+          "type": "integer",
+          "format": "int32",
+          "description": "The number of session hosts migrated by hostpool update."
+        },
+        "sessionHostsRollbackFailed": {
+          "type": "integer",
+          "format": "int32",
+          "description": "The number of session hosts that failed to rollback."
+        }
+      }
+    },
+    "HostPoolUpdateFullProperties": {
+      "description": "Hostpool update full properties including session host config, hostpool update parameters, and migration progress.",
+      "type": "object",
+      "readOnly": true,
+      "properties": {
+        "updateStatus": {
+          "enum": [
+            "Validating",
+            "Scheduled",
+            "InProgress",
+            "Pausing",
+            "Paused",
+            "StartFailed",
+            "UpdateFailed",
+            "Cancelling",
+            "Canceled",
+            "Completed"
+          ],
+          "x-ms-enum": {
+            "name": "HostPoolUpdateStatus",
+            "modelAsString": true,
+            "values": [
+              {
+                "value": "Validating",
```

Comment:
We dont have a validating state - we had discussed this between you, Sheikh and I @datr93

Reply:
@shailahirji I believe when you start initiate you validate the all of the configurations and hpParams and fail on error right? If not i'll remove it.

Reply:
removed